raiden-ts/src/utils/matrix.ts

```diff
@@ -30,22 +32,28 @@ export function yamlListToArray(yml: string): string[] {
  * Given a server name (with or without schema and port), return HTTP GET round trip time
  *
  * @param server - Server name with or without schema
+ * @param httpTimeout - Optional timeout for the HTTP request
  * @returns Promise to a { server, rtt } object, where `rtt` may be NaN
  */
-export async function matrixRTT(server: string): Promise<{ server: string; rtt: number }> {
+export function matrixRTT(
+  server: string,
+  httpTimeout?: number,
+): Observable<{ server: string; rtt: number }> {
   let url = server;
   if (!url.includes('://')) {
     url = `https://${url}`;
   }
   url += `/_matrix/client/versions`;
-  let start = Date.now();
-  try {
-    const resp = await fetch(url);
-    if (resp.status < 200 || resp.status >= 300) throw NaN;
-  } catch (e) {
-    start = NaN; // return will also be NaN
-  }
-  return { server, rtt: Date.now() - start };
+  return defer(() => {
+    const start = Date.now();
+    return fromFetch(url).pipe(
+      map(response => (response.ok ? Date.now() : NaN)),
+      timeout(httpTimeout || Number.MAX_SAFE_INTEGER),
```

Comment:
`9007199254740991` might be too optimistic?

Reply:
This is just a way to disable the timeout conditionally without actually removing the operator from the pipeline in the case it's not provided (it is in the only place we use, but I didn't want to enforce this on the interface).
The alternative to keep the parameter optional would be something like:
```
let ret$ = fromFetch(...).pipe(first operators);
if (httpTimeout) ret$ = ret$.pipe(timeout(httpTImeout));
return ret$.pipe(tail operators);
```
Which is way uglier IMO, and disabling timeouts using large-enough timeouts is very common pattern.